frappe/installer.py

```diff
@@ -777,53 +738,84 @@ def is_downgrade(sql_file_path, verbose=False):
 
 	from semantic_version import Version
 
-	head = "INSERT INTO `tabInstalled Application` VALUES"
+	backup_version = None
+	try:
+		backup_version = extract_version_from_dump(sql_file_path)
+	except Exception:
```

Comment:
Exceptions should not be used for control flow if possible, if the return value is None that indicates better parsing failure? :thinking: